docs/content/rules/sort-classes.mdx

```diff
@@ -215,146 +215,194 @@ Allows you to use comments to separate the class members into logical groups. Th
+- Example: `private-static-accessor-property`.
+
+#### Index-signature
+- Selectors: `index-signature`.
+- Modifiers: No modifier available.
+- Example: `index-signature`.
+
+#### Properties
+- Selectors: `property`.
+- Modifiers: `abstract`, `decorated`, `override`, `readonly`, `protected`, `private`, `public`, `static`.
+- Example: `readonly-decorated-property`.
+
+#### Important notes
+
+##### Scope of the `private` modifier
+The `private` modifier will currently match any of the following:
+- Elements with the `private` keyword.
+- Elements with their name starting with `#`.
+- Elements with their name starting with `_`.
+
+##### Scope of the `public` modifier
+Elements that are not `protected` nor `private` will be matched with the `public` modifier, even if the keyword is not present.
+
+
+##### The `unknown` group
+Members that don’t fit into any other group can be matched with the `unknown` group.
+
+##### Behavior when multiple groups match an element
+
+The lists of selectors and modifiers above are both sorted by importance, from most to least important.
+In case of multiple groups matching an element, the following rules will be applied:
+
+1. Selector priority: `constructor`, `get-method` and `set-method` groups will always take precedence over `method` groups.
+2. If the selector is the same, the group with the most modifiers matching will be selected.
+3. If modifiers quantity is the same, order will be chosen based on modifier importance as listed above.
 
+Example 1:
 ```ts
-class Example {
-  // 'index-signature'
-  [key: string]: any;
+abstract class Class {
 
-  // 'protected-decorated-accessor-property'
-  @SomeDecorator
-  protected get accessor() {
-    return this._value;
-  }
+    protected abstract get field();
 
-  // 'private-decorated-accessor-property'
-  @SomeDecorator
-  private get accessor() {
-    return this._value;
-  }
+}
+```
 
-  // 'decorated-accessor-property'
-  @SomeDecorator
-  get accessor() {
-    return this._value;
-  }
+`field` can be matched by the following groups, from most to least important:
+- `abstract-protected-get-method` or `protected-abstract-get-method`.
+- `abstract-get-method`.
+- `protected-get-method`.
+- `get-method`.
+- `abstract-protected-method` or `protected-abstract-method`.
+- `abstract-method`.
+- `protected-method`.
+- `method`.
 
-  // 'protected-decorated-property'
-  @SomeDecorator
-  protected _value: number;
+Example 2: (The most important group is written in the comments)
 
-  // 'private-decorated-property'
-  @SomeDecorator
-  private _value: number;
+```ts
+abstract class Example extends BaseExample {
+  // 'index-signature'
+  [key: string]: any;
 
-  // 'decorated-property'
+  // 'public-static-property'
+  static instance: Example;
+
+  // 'protected-abstract-override-readonly-decorated-property'
   @SomeDecorator
-  public value: number;
+  protected abstract override readonly _value: number;
```

Comment:
This illustrates the problem I have with `_` in JS: if I'm not mistaken, the rule will consider it as a private name, but because of the `protected` it will be flagged it as `protected`.

As a comparison, `protected #value` would have raised an error.

We shouldn't mix official JS syntax and developer conventions.

Reply:
`_` to be removed